test/prototype_common_utils.py

```diff
@@ -1,28 +1,127 @@
+"""This module is separated from common_utils.py to prevent the former to be dependent on torchvision.prototype"""
+
+import collections.abc
+import enum
 import functools
 import itertools
 
 import PIL.Image
 import pytest
-
 import torch
 import torch.testing
 from torch.nn.functional import one_hot
-from torch.testing._comparison import assert_equal as _assert_equal, TensorLikePair
+from torch.testing._comparison import (
+    assert_equal as _assert_equal,
+    BooleanPair,
+    NonePair,
+    NumberPair,
+    TensorLikePair,
+    UnsupportedInputs,
+)
 from torchvision.prototype import features
-from torchvision.prototype.transforms.functional import to_image_tensor
+from torchvision.prototype.transforms.functional import convert_image_dtype, to_image_tensor
 from torchvision.transforms.functional_tensor import _max_value as get_max_value
 
+__all__ = ["assert_close"]
+
+
+class PILImagePair(TensorLikePair):
```

Comment:
This is a superset of what the old `ImagePair` did. It includes the options to only test the aggregated difference or check the percentage of differing pixels. That is on par with what we are currently doing in our stable functional tests:

https://github.com/pytorch/vision/blob/a67cc87a33a3f713aebf5299bdeb2672c98e0bc5/test/common_utils.py#L172-L174